### PR TITLE
[SPARK-44528][CONNECT] Support proper usage of hasattr() for Connect dataframe

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
Currently Connect does not allow the proper usage of Python's `hasattr()` to identify if an attribute is defined or not. This patch fixes that bug (it's working in regular PySpark).

### Why are the changes needed?
Bugfix

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by grundprinzip** [comment]
@HyukjinKwon PTAL please.

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 1]
```diff
@@ -157,6 +157,21 @@ def spark_connect_clean_up_test_data(cls):
...
+        sdf = self.spark.range(10)
+
+        sdf._simple_extension = 10
+        cdf._simple_extension = 10
+
+        self.assertEqual(sdf._simple_extension, cdf._simple_extension)
+        self.assertEqual(type(sdf._simple_extension), type(cdf._simple_extension))
+
+        self.assertTrue(hasattr(cdf, "_simple_extension"))
+        self.assertFalse(hasattr(cdf, "_simple_extension_does_not_exsit"))
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:172]: argh, it needs reformat script run

File: python/pyspark/sql/connect/dataframe.py [hunk 2]
```diff
@@ -1584,8 +1584,16 @@ def __getattr__(self, name: str) -> "Column":
                 error_class="NOT_IMPLEMENTED",
                 message_parameters={"feature": f"{name}()"},
             )
+
+        if name not in self.columns:
```
Comment (beliefer) [inline python/pyspark/sql/connect/dataframe.py:1588]: How about use `elif:` here ?

Comment (HyukjinKwon) [reply python/pyspark/sql/connect/dataframe.py:1588]: yeah but this is matched with the non Spark Connect side. I think it's just fine to leave it matched.

Comment (beliefer) [reply python/pyspark/sql/connect/dataframe.py:1588]: I checked the pyspark. As you said, I'm +1 for you comment.

**Comment by grundprinzip** [comment]
@HyukjinKwon I fixed the remaining failing test. PTAL again :) Thanks

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.

**Pull request closed without merging**